delta/sdk-views/src/main/scala/ch/epfl/bluebrain/nexus/delta/sdk/views/indexing/ProjectionProgressStreamOps.scala

```diff
@@ -0,0 +1,69 @@
+package ch.epfl.bluebrain.nexus.delta.sdk.views.indexing
+
+import ch.epfl.bluebrain.nexus.delta.rdf.IriOrBNode.Iri
+import ch.epfl.bluebrain.nexus.delta.sdk.views.indexing.ProjectionProgressStreamOps.metricsPrefix
+import ch.epfl.bluebrain.nexus.delta.sdk.views.model.ViewIndex
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionProgress
+import ch.epfl.bluebrain.nexus.delta.sourcing.projections.ProjectionStream.StreamOps
+import fs2.Stream
+import kamon.Kamon
+import kamon.tag.TagSet
+import monix.bio.Task
+import monix.execution.atomic.AtomicLong
+
+/**
+  * Provides extensions methods for fs2.Stream[Message] to implement view statistics
+  */
+class ProjectionProgressStreamOps[A](val stream: Stream[Task, ProjectionProgress[A]]) extends StreamOps[A] {
+
+  /**
+    * Record view metrics in Kamon
+    * @param view           the view
+    * @param viewType       the view type
+    * @param additionalTags additional Kamon tags
+    */
+  def viewMetrics(
+      view: ViewIndex[_],
+      viewType: Iri,
+      additionalTags: Map[String, Any] = Map.empty
+  ): Stream[Task, ProjectionProgress[A]] = {
+    val defaultTags                      = Map(
+      "project"      -> view.projectRef.toString,
+      "organization" -> view.projectRef.organization.toString,
+      "viewId"       -> view.id.toString,
+      "type"         -> viewType.toString
+    )
+    val tags                             = defaultTags ++ additionalTags
+    val processedEventsGauge             = Kamon
+      .gauge(s"${metricsPrefix}_gauge_processed_events")
+      .withTags(TagSet.from(tags))
+    val processedEventsCounter           = Kamon
+      .counter(s"${metricsPrefix}_counter_processed_events")
+      .withTags(TagSet.from(tags))
+    val processedEventsCount: AtomicLong = AtomicLong(0)
```

Comment:
can this pose problems when the stream is restarted (service restart or shard balancing)?